Added support for macOS/Linux

fnky opened the pull request:
- Added pragmas in `display.cpp` for OS detection
  - Uses `fopen` instead of `fopen_s` for macOS & Linux
- Makefile changed to support clang (macOS)
  - Removed unsupported compiler/linker flags for clang

fnky committed:
Added support for macOS/Linux

Harha reviewed (changes requested):
Correct me if I'm wrong, but you have 'ifneq ($(OS), Windows_NT) ifeq ($(shell uname -s), Darwin)' twice in there.

Also '-static-libgcc -static-libstdc++' gets added for both Windows and OSX? So that does not make much sense to me at least.

Maybe correct it so that this all makes more sense, thanks! It works though, as is.

Harha commented:
Also a better place for those OS_WINDOWS, OS_MACOS and OS_LINUX macros would probably be ./src/macros.h

Thank you a lot in advance! I can merge this and do the further changes I want myself if you wish to not edit your pull request any more.

fnky commented:
The double OS check for the warning flags were my slip up.

However the second is because `uname` is assumed to not exist on Windows and `$(OS)` didn't work on Mac/Linux. It's one of those Makefile buggers :(

fnky committed:
Remove double OS check

fnky committed:
Move OS pragmas to macros

fnky committed:
Add support to init for Unix

- Added mkdir commands with correct separator for Unix-like systems
- Fixed copy command to be recursive

fnky commented:
Added a few bits and pieces to Makefile for `make init`, since it didn't work as expected. I don't know wether Windows doesn't care about forward/backward slash in directory paths, but POSIX does not like it.

If this could be simplified, e.g. just using forward slash (`/`) that would be better, but I don't know how Windows would behave in such a case.
